src/library/baseplaylistfeature.cpp

```diff
@@ -650,22 +649,23 @@ QModelIndex BasePlaylistFeature::constructChildModel(int selected_id) {
     return m_childModel.index(selected_row, 0);
 }
 
-void BasePlaylistFeature::updateChildModel(int selected_id) {
-    buildPlaylistList();
+void BasePlaylistFeature::updateChildModel(int changedPlaylistId) {
+    // the following call fetches whole m_playlistList.
+    // Anything may have change due to a concurrent Mixxx instance.
+    updatePlaylistLabel(changedPlaylistId);
```

Comment:
The parameter `changedPlaylistId` is too verbose. This functions doesn't need to care what happened to the playlist before. It is instructed to update the tree item of a particular `playlistId` and that's it.

The functions should actually be named `updatePlaylistItem()`, because it updates the entire item and not only the label.

The code is still more complex than it needs to be:
```
QString updatedLabel = updatePlaylistLabel(playlistId);
TreeItem* playlistItem = findPlaylistItem(playlistId);
VERIFY_OR_DEBUG_ASSERT(playlistItem) {
   return;
}
DEBUG_ASSERT(playlistItem->getData() == playlistId);
item->setLabel(updatedLabel);
decorateChild(playlistItem, playlistId);
```

Reply:
Is `m_playlistItem` actually needed if all this information is stored redundantly in the `TreeItem`s?

Reply:
See also: `CrateFeature` and `CrateSummary`, no redundant and error-prone extra bookkeeping needed.